site/en/articles/webdriver-bidi-2023/index.md

```diff
@@ -0,0 +1,109 @@
+---
+title: 'WebDriver BiDi: 2023 status update'
+description: >
+  This article gives an overview of what’s new in WebDriver BiDi in 2023.
+layout: layouts/blog-post.njk
+authors:
+  - mathiasbynens
+date: 2023-05-09
+updated: 2023-05-09
+hero: 'image/dPDCek3EhZgLQPGtEG3y0fTn4v82/nrI18Gn6bcbje10UnJBj.jpg'
+alt: ''
+tags:
+  - test-automation
+  - devtools
+---
+This article gives an overview of what’s new in WebDriver BiDi in 2023.
+
+## What is WebDriver BiDi?
+
+WebDriver is a browser automation protocol, defined as a W3C standard, with implementations in ChromeDriver, GeckoDriver, and WebKitDriver.
+
+Chromium also has its own proprietary browser automation protocol: the Chrome DevTools Protocol, or CDP.
+
+There are some fundamental differences between these two protocols: WebDriver is an interoperable standard, but the protocol is less efficient and lacks features that CDP has. In contrast, CDP is more efficient and powerful, but less interoperable.
+
+That’s why in 2020, [the W3C Browser Testing and Tools Working Group](https://www.w3.org/groups/wg/browser-tools-testing) began work on [WebDriver BiDi](https://w3c.github.io/webdriver-bidi/), a new standard browser automation protocol that bridges the gap between the WebDriver “Classic” and CDP protocols. The best of both worlds! Read [_“A look back in time: the evolution of test automation”_](/blog/test-automation-evolution/) and [_“WebDriver BiDi — the future of cross-browser automation”_](/articles/webdriver-bidi/) for more background.
+
+The WebDriver BiDi effort involves standardization work, the creation of Web Platform Tests, and implementations for different browser engines.
+
+## Where are we now?
+
+In 2022, both [Chrome/ChromeDriver 106](https://crbug.com/chromedriver/4016) and [Firefox 102](https://bugzilla.mozilla.org/show_bug.cgi?id=1753997) shipped support for the WebDriver BiDi standard.
+
+Since then, WebDriver BiDi gained adoption in popular frameworks, addressing top developer pain points by unlocking highly requested features such as [logging support](https://www.selenium.dev/documentation/webdriver/bidirectional/bidirectional_w3c/log/).
+
+## Logging powered by WebDriver BiDi
+
+A common use case is to automatically verify that a web page loads without any console logs, warnings, or errors and without any uncaught JavaScript exceptions. Another use case is that you may wish to write an automated test to check that a web page logs a warning or throws an exception when intended. These problems can’t be solved using WebDriver “Classic” because [it’s not bidirectional](/blog/test-automation-evolution/#what-are-the-low-level-controls). WebDriver BiDi now makes this possible.
```

Comment:
```suggestion
A common use case is to automatically verify that a web page loads without any console logs, warnings, or errors and without any uncaught JavaScript exceptions. Another use case is that you may wish to write an automated test to check that a web page logs a warning or throws an exception when intended. These problems can’t be solved using WebDriver Classic because [it’s not bidirectional](/blog/test-automation-evolution/#what-are-the-low-level-controls). WebDriver BiDi now makes this possible.
```